### PR TITLE
Instead of rarfile use unrar

### LINKED ISSUE #131: ModuleNotFoundError: No module named 'launcher.common'
With 3a4554e8e9b3b84823850f57d236bdb03279fc79
gamma-launcher fails with:
Traceback (most recent call last):
  File "/home/sobkas/gamma/bin/gamma-launcher", line 5, in <module>
    from launcher.cli import main
  File "/home/sobkas/gamma/lib/python3.11/site-packages/launcher/cli.py", line 2, in <module>
    from launcher.commands import (
  File "/home/sobkas/gamma/lib/python3.11/site-packages/launcher/commands/__init__.py", line 1, in <module>
    from .check import CheckAnomaly, CheckMD5
  File "/home/sobkas/gamma/lib/python3.11/site-packages/launcher/commands/check.py", line 7, in <module>
    from launcher.common import anomaly_arg, gamma_arg
ModuleNotFoundError: No module named 'launcher.common'

### PR DESCRIPTION
**Pull request by sobkas**
Resolve #131

### CODE REVIEW

File: launcher/archive.py [hunk 1]
```diff
@@ -33,7 +33,8 @@ def get_mime_from_file(filename) -> str:
 
 def _7zip_extract(f: str, p: str) -> None:
     if run(['7z', 'x', '-y', f'-o{p}', f]).returncode != 0:
-        raise SevenZipDecompressionError(f'7z error will decompressing {f}')
+        if run(['unrar', '-y ','x', f'{f}', f'{p}']).returncode != 0:
```
Comment (Mord3rca) [inline launcher/archive.py:36]: This will use unrar instead of 7z for all archives (win32) and rar / 7z+bcj2 for *nix.

**Comment by Mord3rca** [comment]
The good way to do it is to create a new function and check `rarfile` configuration to use the correct executable for unrar decompression.

```python
import rarfile

a = rarfile.tool_setup()   # This will get the tool use

# To check it use (in a dict ?)
a.setup == rarfile.UNRAR_CONFIG  # Use unrar
a.setup == rarfile.BSDTAR_CONFIG  # Use bsd
....
```

File: launcher/archive.py (hunk 1 above)
Comment (sobkas) [reply launcher/archive.py:36]: It will use unrar for all archives that fail to extract.

Comment (sobkas) [reply launcher/archive.py:36]: I'm thinking about checking if filename contains .rar and then only use unrar if it's true

**Comment by sobkas** [comment]
> The good way to do it is to create a new function and check `rarfile` configuration to use the correct executable for unrar decompression.
> 
> ```python
> import rarfile
> 
> a = rarfile.tool_setup()   # This will get the tool use
> 
> # To check it use (in a dict ?)
> a.setup == rarfile.UNRAR_CONFIG  # Use unrar
> a.setup == rarfile.BSDTAR_CONFIG  # Use bsd
> ....
> ```

But all it does is:
UNRAR_TOOL = "unrar"
and then uses it to extract files
UNRAR_CONFIG = {
    "open_cmd": ("UNRAR_TOOL", "p", "-inul"),
    "check_cmd": ("UNRAR_TOOL", "-inul", "-?"),

File: launcher/archive.py (hunk 1 above)
Comment (Mord3rca) [reply launcher/archive.py:36]: Creating a new function to fix correctly #127 (and reverting workaround #128) will be better

**Comment by Mord3rca** [comment]
The idea is to detect which command is used by `rarfile`, and then you exec your shell command according to rarfile's detection

**Comment by sobkas** [comment]
> The idea is to detect which command is used by `rarfile`, and then you exec your shell command according to rarfile's detection

And there is option of not using rarfile, there are other python libraries that aren't a failure...

**Comment by sobkas** [comment]
So any toughs?

File: README.md [hunk 2]
```diff
@@ -110,20 +110,11 @@ Also remove some shaders mods:
 
 The `distutils` module is required to install Python packages but it was removed in Python 3.12. You can still use it by installing `setuptools` (inside the venv): `pip install setuptools`
 
-### rarfile.RarCannotExec: Cannot find working tool
```
Comment (Mord3rca) [inline README.md:113]: This is still possible on previous version of the tool and should not be removed

File: README.md [hunk 3]
```diff
@@ -110,20 +110,11 @@ Also remove some shaders mods:
...
-
-You are missing a tool that extracts RAR files. You can use something like `unrar` on Linux. To install it run the following command:
-- On Ubuntu/Debian: `sudo apt install unrar`
-- On Fedora **(RPM Fusion required)**: `sudo dnf install unrar`. Note that before the installation you will need to enable the **non-free** RPM Fusion repository since the default package provided by Fedora will cause problems. [Instructions on how to enable it here](https://rpmfusion.org/Configuration).
-- On Arch/Manjaro: `sudo pacman -S unrar`
-
-### SevenZipDecompressionError
```
Comment (Mord3rca) [inline README.md:120]: Same here, previous versions can still except with this

File: README.md [hunk 4]
```diff
@@ -110,20 +110,11 @@ Also remove some shaders mods:
...
-
-7zip seems to have trouble with this archive, if this is a rar file, make sure to install `7z` with rar capabilities.
-- On Ubuntu/Debian: `sudo apt install 7zip-rar`
-
-You can use `rarfile` module for decompressing by setting **GAMMA_LAUNCHER_USE_RARFILE** env.
-Warning: This may be slow for some archive
+### LookupError: Couldn't find path to unrar library.
+You are missing a library that extracts RAR files. You can use something like `libunrar` on Linux. To install it run the following command:
+- On Ubuntu/Debian: `sudo apt install libunrar5t64` or `sudo apt install libunrar5`
```
Comment (Mord3rca) [inline README.md:115]: `libunrar5t64` is only for Ubuntu 24.04 LTS (Noble Numbat)
see: https://pkgs.org/search/?q=libunrar

File: launcher/archive.py [hunk 5]
```diff
@@ -2,18 +2,14 @@
 from pathlib import Path
 from tempfile import TemporaryDirectory
 
-from os import getenv, name as os_name
+from os import name as os_name
 from subprocess import run
 from py7zr import SevenZipFile
-from rarfile import RarFile
+from unrar import rarfile
```
Comment (Mord3rca) [inline launcher/archive.py:8]: `from unrar.rarfile import RarFile`

**Review by Mord3rca — CHANGES_REQUESTED** [review]
Can you use `git revert` to revert 3d39a0db9bfd87346f2cf39e5c15590fbcdc75b7 & 22a1820b33c3936525e7177944b9355e1832eb90 (PR #128) and then do your changes ?

So commit history will be clearer, thanks

File: README.md (hunk 4 above)
Comment (sobkas) [reply README.md:115]: Debian have it too

**Comment by sobkas** [comment]
Ok, fixed

**Comment by sobkas** [comment]
Btw. I have plans for:
1. Adding support for custom Gamma repo to full-install(implemented)
2. GUI, based on GTK4, still learning GTK4.(wip)

**Comment by Mord3rca** [comment]
Small flake8 defects and I'll merge then :) Thanks for the work !

**Comment by Mord3rca** [comment]
> Btw. I have plans for:
> 
>     1. Adding support for custom Gamma repo to full-install(implemented)
> 
>     2. GUI, based on GTK4, still learning GTK4.(wip)

Yeah, it will me a good idea if we want to put Linux specific patches in there.

And I'm 100% ok for the GUI but it need to be optional in the installation phase

**Review by Mord3rca — APPROVED** [review]

**Pull request merged**